src/display.cpp

```diff
@@ -827,7 +827,7 @@ bool display::screenshot(const std::string& filename, bool map_screenshot)
 #endif
 
 		//NOTE: need to be sure that we free this huge surface (is it enough?)
-		map_screenshot_surf_ = nullptr;
```

Comment:
Considering all the `foo = nullptr` lines, we might take a page out of `std::unique_ptr`'s book and have an implicit `surface(nullptr_t)` constructor.

Reply:
We do. `surface::surface() : surface(nullptr) {}`

Reply:
The problem is I didn't want to change the flow of working code to default construct and only copy-assign when it's non-default.

Reply:
No, I mean `surface(nullptr_t) : surface_(nullptr) {} //implicit`